platform/http_request.cpp

```diff
@@ -386,6 +387,12 @@ HttpRequest::~HttpRequest()
 {
 }
 
+HttpRequest::ProgressT const & HttpRequest::Progress() const
+{
+  ASSERT(m_progress.first <= m_progress.second, ());
```

Comment:
Значит в коде, который использует прогресс, нужно добавить проверку на < 0 (случай, когда размер загружаемого файла не известен).

> On Apr 7, 2016, at 10:07, Илья Гречухин <notifications@github.com> wrote:
> 
> In platform/http_request.cpp <https://github.com/mapsme/omim/pull/2756#discussion_r58827307>:
> 
> > @@ -386,6 +387,12 @@ HttpRequest::~HttpRequest()
> >  {
> >  }
> >  
> > +HttpRequest::ProgressT const & HttpRequest::Progress() const
> > +{
> > +  ASSERT(m_progress.first <= m_progress.second, ());
> -1 нигде в коде не проверяется, но лишь приводит к блуждающим ошибкам.
> Без проверки на -1, это просто намеренное добавление ошибки в приложение.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2756/files/193bf11a2e6f8a3ead0d909e7f91c3f351ef2e0a#r58827307>